content/Journey/_index.md

```diff
@@ -0,0 +1,98 @@
+{{%/alert%}}
+
+Excerpt from *Accelerate* by Nicole Forsgren PhD, Jez Humble & Gene Kim
+
+## Solving the Challenge of CI
+
+The first challenge for the team is [continuous integrations](https://www.martinfowler.com/articles/continuousIntegration.html). CD requires CI and CI is very effective at uncovering most of the problems many teams have that impact quality. Martin Fowler has an [excellent blog post on introducing CI](https://www.martinfowler.com/articles/continuousIntegration.html#IntroducingContinuousIntegration) into the workflow. Over the years, we've seen many common problems that teams have.
+
+You may find more. Notice that tooling is rarely the problem. Always, "why can't we deliver working changes to the trunk today?", is the roadmap of problems to solve.  
+
+### Code review takes too long / has too many approvers
+
+Having more than one reviewer on a PR is a process smell for issues with the quality process. If one reviewer isn't enough, ask why. Those are defects. Code review, if we aren't pair / mob programming, should be minutes of effort, not hours. The changeset should be small, include all expected tests, and most of the heavy lifting of traditional code review should be automated with linters and formatters. The code review process adds drag to the process that encourages larger changes. We want to keep changes small. Focus on improving the flow fo communication to reduce the drag.
+
+- Worst: Code reviews are done solo and comments sent back to the developer. This adds the most drag.
+- Less bad: The reviewer and author meet to review and correct issues. There is still a wait time for that meeting, but it is more efficient.
+- Best: [Pair programming](https://martinfowler.com/bliki/PairProgramming.html) builds code review into the flow.
+
+### Tests are deferred or skipped
+
+We must treat tests as first-class citizens. They are more important than the code we are testing. They cannot be an afterthought and we need to make sure we are using them to increase our confidence rather than meeting some arbitrary ["coverage metric"](https://dojoconsortium.org/metrics/code-coverage/). We don't need all of the legacy code to be 100% tested, or even 1% tested to start CI. We need to commit to "we will never push untested changes." Timelines are irrelevant if we deliver things on time that don't work.
+
+### The team lacks knowledge on how to write tests for CI
+
+Dig into testing. Learn effective test patterns. Many people start out testing implementation and then struggle to keep tets current as implementation changes. Avoid implementation testing. Test behaviors. If we cannot refactor code without changing the test then we are testing incorrectly. Focus on BDD to define tests and TDD to implement tests. People often push back on TDD, but that is most commonly because they never learned how to do it correctly. BDD was created to help. Lean in.
```

Comment:
```suggestion
Dig into testing. Learn effective test patterns. Many people start out testing implementation and then struggle to keep tests current as implementation changes. Avoid implementation testing. Test behaviors. If we cannot refactor code without changing the test then we are testing incorrectly. Focus on BDD to define tests and TDD to implement tests. People often push back on TDD, but that is most commonly because they never learned how to do it correctly. BDD was created to help. Lean in.
```